kymatio/scattering2d/core/scattering2d.py

```diff
@@ -0,0 +1,56 @@
+# Authors: Edouard Oyallon
+# Scientific Ancestry: Edouard Oyallon, Laurent Sifre, Joan Bruna
+
+
+__all__ = ['scattering2d']
+
+def scattering2d(x, pad, unpad, backend, J, L, phi, psi, max_order):
+    subsample_fourier, modulus, fft, cdgmm, finalize = backend.subsample_fourier, backend.modulus,\
+                                                         backend.fft, backend.cdgmm, backend.finalize
+    order0_size = 1
+    order1_size = L * J
+    order2_size = L ** 2 * J * (J - 1) // 2
+    output_size = order0_size + order1_size
+
+    if max_order == 2:
+        output_size += order2_size
+
+    out_S_0, out_S_1, out_S_2 = [], [], []
```

Comment:
So - I think it's ok. Like, it's a list/array of pointer, not the actual memory used to store the array- i'm open to anything else

Reply:
I ignored this. Please, fix.

Reply:
Ok will take a look. Should require a new backend primitive, `new` to create an array of a given size.

Reply:
(I'm slightly surprised this will make a significant difference)

Reply:
but thanks for taking care ofit!!

Reply:
I'm getting a 10% reduction in speed for a 16×256×256 array with J=6 on my laptop (CPU) between master and the current version. Might be something else, but it's worth looking into. A memory movement saved is a memory movement earned.

Reply:
Ok. So I've a proposed solution in 0b996ed2c4a1310b31ed12b5f10de5268afd9b15. Let me know what you think. I'm able to get the same performance as before for the example above.

Reply:
It looks good, I'm surprised that torch has no empty-like.

Reply:
I think it does. I just copy-pasted the code I found in `scattering2d`. Since `new` isn't documented, I wasn't sure if it was safe to replace with something else (like `torch.empty_like` or just `torch.empty`).

Reply:
Anyhow. Resolving this.